### PR TITLE
gh-104773: PEP 594: Remove the spwd module

### PR DESCRIPTION
**Pull request by vstinner**
Remove spwd from the configure script and Modules/Setup.

* Issue: gh-104773

----
:books: Documentation preview :books:: https://cpython-previews--104871.org.readthedocs.build/

### CODE REVIEW

**Review by hugovk — APPROVED** [review]

File: configure.ac [hunk 1]
```diff
@@ -7085,12 +7085,11 @@ dnl Modules that are not available on some platforms
 dnl AIX has shadow passwords, but access is not via getspent()
```
Comment (arhadthedev) [inline configure.ac]: This line still mentions "shadow passwords" (the removed `spwd` is *Shadow Password Database*).

File: Modules/Setup.stdlib.in [hunk 2]
```diff
@@ -120,7 +120,6 @@
 @MODULE_SELECT_TRUE@select selectmodule.c
 @MODULE__SOCKET_TRUE@_socket socketmodule.c
 # AIX has shadow passwords, but does not provide getspent API
```
Comment (arhadthedev) [inline Modules/Setup.stdlib.in]: The same.

File: configure.ac (hunk 1 above)
Comment (vstinner) [reply configure.ac]: Oh right. I also removed it.

**Comment by vstinner** [comment]
@arhadthedev: I updated my PR to remove the two mentions of AIX shadow passwords.

**Review by arhadthedev — APPROVED** [review]

**Pull request merged**